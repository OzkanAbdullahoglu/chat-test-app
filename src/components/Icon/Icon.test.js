/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import SvgIcon from './Icon';


describe('Icon', () => {
  const props = {
    incomingIcon: true,
    incomingAvatar: true,
    sendIcon: true,
    scrollDown: true,
  };

  it('renders without crashing', () => {
    const icon = shallow(<SvgIcon {...props} />);
    expect(toJson(icon)).toMatchSnapshot();
  });
  it('renders properly under default props', () => {
    const icon = shallow(<SvgIcon {...props} />);
    expect(icon).toHaveStyleRule('height', '1.5em');
    expect(icon).toHaveStyleRule('width', '1.5em');
    expect(icon).toHaveStyleRule('font-size', '1.5rem');
    expect(icon).toHaveStyleRule('color', '#00bcd4');
    expect(icon).toHaveStyleRule('margin-right', '0.1rem');
    expect(icon).toHaveStyleRule('position', 'absolute');
    expect(icon).toHaveStyleRule('transform', 'rotate(90deg)');
  });
  it('renders properly under custom props', () => {
    const customProps = {
      incomingIcon: false,
      incomingAvatar: false,
      sendIcon: false,
      scrollDown: false,
    };
    const icon = shallow(<SvgIcon {...customProps} />);
    expect(icon).toHaveStyleRule('height', '2em');
    expect(icon).toHaveStyleRule('width', '2em');
    expect(icon).toHaveStyleRule('font-size', '1rem');
    expect(icon).toHaveStyleRule('color', '#fff');
    expect(icon).toHaveStyleRule('margin-right', '0');
    expect(icon).toHaveStyleRule('position', 'relative');
    expect(icon).toHaveStyleRule('transform', 'rotate(0)');
  });
});

// still we have to do...
