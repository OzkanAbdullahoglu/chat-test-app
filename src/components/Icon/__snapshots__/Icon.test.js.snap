// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders without crashing 1`] = `
.c0 {
  fill: currentColor;
  width: 1.5em;
  height: 1.5em;
  font-size: 1.5rem;
  color: #00bcd4;
  margin-right: 0.1rem;
  position: absolute;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.sc-AxjAm:hover .c0 {
  color: #fff;
}

<svg
  className="c0"
/>
`;
